File: mobile/lib/domain/entities/trip_entity.dart
/// @file trip_entity.dart
/// @description 行程領域模型 / Trip domain entity
/// @description_zh 定義行程的核心業務資料結構
/// @description_en Defines the core business data structure of a trip
library;

class TripEntity {
  final String id;
  final String title;
  final String? location;
  final DateTime startDate;
  final DateTime endDate;
  final int memberCount;
  final String? iconName;
  final int? colorValue;
  final String status;

  TripEntity({
    required this.id,
    required this.title,
    this.location,
    required this.startDate,
    required this.endDate,
    this.memberCount = 1,
    this.iconName,
    this.colorValue,
    this.status = 'ACTIVE',
  });

  // 輔助方法：計算旅程天數 / Helper: Calculate trip duration in days
  int get durationDays => endDate.difference(startDate).inDays + 1;
}

File: backend/src/modules/trips/entities/trip.entity.ts
import { Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn } from 'typeorm';

/**
 * @file trip.entity.ts
 * @description 行程實體 / Trip Entity
 * @description_zh 定義行程在資料庫中的表結構
 * @description_en Defines the table structure for trips in the database
 */

@Entity('trips')
export class Trip {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ length: 100 })
  title: string;

  @Column({ length: 255, nullable: true })
  location: string;

  @Column({ type: 'timestamp' })
  startDate: Date;

  @Column({ type: 'timestamp' })
  endDate: Date;

  @Column({ default: 1 })
  memberCount: number;

  @Column({ type: 'uuid', unique: true, default: () => 'uuid_generate_v4()' })
  share_token: string;

  @Column({ length: 50, nullable: true })
  icon_name: string;

  @Column({ type: 'bigint', nullable: true })
  color_value: number;

  @Column({ length: 50, nullable: true })
  owner_id: string;

  @Column({ length: 20, default: 'ACTIVE' })
  status: string;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;
}
